xbmc/network/WebServer.cpp

```diff
@@ -765,6 +766,10 @@ int CWebServer::CreateFileDownloadResponse(const std::shared_ptr<IHTTPRequestHan
   std::shared_ptr<XFILE::CFile> file = std::make_shared<XFILE::CFile>();
   std::string filePath = handler->GetResponseFile();
 
+  // white/black list access check
+  if (!CFileUtils::ZebraListAccessCheck(filePath))
+    return SendErrorResponse(request.connection, MHD_HTTP_NOT_FOUND, request.method);
```

Comment:
Since I'm still notified for that PR.

Currently https://github.com/xbmc/xbmc/blob/master/xbmc/network/httprequesthandler/HTTPVfsHandler.cpp#L84 returns 401 for sources / files that are not allowed for remote access.

This is what is used by most remotes to help the user configuring correctly it's sources when having a problem.

Not talking again about the fact that this patch removes access to allowed sources (@Montellese in case you missed that side effect), but if everything returns 404, it may worth more than a comment in code but some wider communication for all users to be aware of the change.

And if this is merged and access to sources is added back later but still returns 404 then maybe extends the JSON getSources to publish the allowsharing so that there's enough information to help users.